x-pack/plugins/security_solution/common/search_strategy/security_solution/matrix_histogram/events/index.ts

```diff
@@ -26,10 +26,12 @@ export interface EventsActionGroupData {
   doc_count: number;
 }
 
+export type Fields = Record<string, unknown[] | Fields[]>;
```

Comment:
this type is referring to itself is that kosher? does it recurse? broke my brain a little bit

Reply:
Yep, it's recursive! This was the most accurate representation of nested fields (and thus "fields" in general) I could think of. It appears to work with all our existing uses of the property without any changes  because `unknown[]` is greedier than `Fields[]`, but e.g. I needed to [narrow](https://github.com/elastic/kibana/pull/96187/files#diff-d843685d4f1dd2b3156e080a3800fff8459859eb43ba9b3df669a2456f3b1ffeR107) in cases where we know it's gong to be the recursive form.